eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Deleg.hs

```diff
@@ -0,0 +1,255 @@
+      !(Credential 'Voting (EraCrypto era)) -- Credential which is not registered
+
+  | RegDelegCertDRepAlreadyRegisteredDELEG
+      !(Credential 'Staking (EraCrypto era)) -- Credential which is already registered
+  | WrongCertificateTypeDELEG -- The DCertPool constructor should not be used by this transition
+  deriving (Generic)
+
+instance 
+  NoThunks (PredicateFailure (EraRule "DELEG" era)) => 
+  NoThunks (ConwayDelegPredFailure era)
+deriving instance
+  Eq (PredicateFailure (EraRule "DELEG" era)) =>
+  Eq (ConwayDelegPredFailure era)
+deriving instance
+  Show (PredicateFailure (EraRule "DELEG" era)) =>
+  Show (ConwayDelegPredFailure era)
+
+-- newtype ConwayDelegEvent era = NewEpoch EpochNo
+newtype ConwayDelegEvent era = DelegEvent (Event (EraRule "DELEG" era))
+
+instance 
+  ( EraTx era
+  , ConwayEraTxBody era
+  , State (EraRule "DELEG" era) ~ CertState era
+  , Signal (EraRule "DELEG" era) ~ TxCert era
+  , TxCert era ~ ConwayTxCert era
+  , Environment (EraRule "DELEG" era) ~ DelegEnv era
+  , EraRule "DELEG" era ~ ConwayDELEG era
+  ) => 
+  STS (ConwayDELEG era) where
+  type State (ConwayDELEG era) = CertState era
+  type Signal (ConwayDELEG era) = TxCert era
+  type Environment (ConwayDELEG era) = DelegEnv era
+  type BaseM (ConwayDELEG era) = ShelleyBase
+  type PredicateFailure (ConwayDELEG era) = ConwayDelegPredFailure era
+  type Event (ConwayDELEG era) = ConwayDelegEvent era
+
+  transitionRules = [conwayDelegTransition @era]
+
+conwayDelegTransition :: 
+  forall era.
+  ( EraTx era
+  -- , ConwayEraTxBody era
+  -- , ShelleyEraTxBody era
+  -- , State (EraRule "DELEG" era) ~ CertState era
+  -- , Signal (EraRule "DELEG" era) ~ TxCert era
+  , TxCert era ~ ConwayTxCert era
+  -- , Environment (EraRule "DELEG" era) ~ DelegEnv era
+  -- , EraRule "DELEG" era ~ ConwayDELEG era
+  ) => 
+  TransitionRule (ConwayDELEG era)
+conwayDelegTransition = do
+  TRC 
+    ( DelegEnv _slot ptr _acnt pp
+    , cState@CertState
+        { certVState = VState{vsDReps}
+        , certPState = _
+        , certDState = certDState@DState{dsUnified} 
+        } 
+    , c
+    ) <- judgmentContext
+  let pd = pp ^. ppKeyDepositL
+  case c of
+    ConwayTxCertDeleg (ConwayRegCert stakeCred sMayCoin) -> do
+      sMayCoin == SNothing || sMayCoin == SJust pd
+        ?! RegCertProvidedCoinNotEqualToExpectedPoolDepositDELEG sMayCoin
+      UM.notMember stakeCred (RewDepUView dsUnified) 
+        ?! RegCertStakeKeyAlreadyRegisteredDELEG stakeCred
+      let ppDeposit = UM.compactCoinOrError pd
+          newRDs = UM.RewDepUView dsUnified UM.∪ (stakeCred, UM.RDPair (UM.CompactCoin 0) ppDeposit)
+          newUMap = UM.PtrUView newRDs UM.∪ (ptr, stakeCred)
+      pure $ 
+        cState 
+          { certDState = 
+              certDState 
+                { dsUnified = newUMap }
+          }
+    ConwayTxCertDeleg (ConwayUnRegCert stakeCred sMayCoin) -> do
+      sMayCoin == SNothing || sMayCoin == SJust pd
```

Comment:
We haven't really put this in the spec yet, but I think the deposit should be checked against the deposit that was actually paid, not the deposit that's currently required by PParams.

Reply:
```suggestion
      sMayCoin == SNothing
```
Since you already check the deposit against the deposit that was actually paid, you don't actually have to check against the deposit required by PParams

Reply:
Noted! Nice catch! :slightly_smiling_face: